### ISSUE TITLE
Expose synthetic switch to enable/disable camera recording

### ISSUE THREAD

**Issue by Ghazgkull** [hjdhjd/homebridge-unifi-protect#651]
**Is your feature request related to a problem? Please describe:**
Unifi Protect allows us to enable/disable recording for each camera, but only in a totally static way. Each camera either records or it doesn't. If it is possible for homebridge-ui-protect to enable/disable recording on Unifi cameras, it would allow Homebridge users to enable/disable recording via Homekit automations.

For example, we could set up automations to enable recording only we're not home. Or during certain times of day. Or to turn off recording when another accessory is controlled.

The use-case that really motivates this request for me is cameras in my home, where I don't want the camera recording my family during the day when we're home. That would be creepy. But I'd like to have my cameras recording at night when we're asleep or when we're not home.

**Describe the solution you'd like:**
I'd like to see an option to expose a switch for each camera that would enable/disable recording. Similar to the switch that's currently provided to enable/disable motion detection.

**Describe alternatives you've considered:**
The Unifi Protect app allows us to configure cameras to only record when motion is detected, but that doesn't satisfy my use-case.

**Additional context:**
I've forked both this repo and the unifi-protect api project to see if I might be able to get a quick sense of how to implement such a feature. I see a `recordingSettings.mode` property in the `ProtectCameraConfigInterface` here: https://github.com/hjdhjd/unifi-protect/blob/6d76ac161b166f9179a0da95d11e277084ab38f7/src/protect-types.ts#L304. Is that a property that can be modified via the API to enable/disable recording?

**Comment by hjdhjd** [comment]
I have plans to add something along these lines in an upcoming update. Messing with the implied contract that HKSV and Home hubs are expecting is more challenging than you might think. :smile: Thanks...and stay tuned!

**Comment by hjdhjd** [comment]
Sorry, I don't maintain my repo that way. It will be in an upcoming release. I'll mention it on this closed issue when it appears. Thanks.

**Comment by hjdhjd** [comment]
Implemented in v5.2.0. Enjoy.

**Comment by Ghazgkull** [comment]
@hjdhjd Wow, awesome. This looks like it will be a great option for folks who are using HKSV!

For those of us schlubs who are out here using local video storage (UCK-G2+, UNVR, etc) instead of cloud storage, could we consider adding an option like the one requested here? Would it be possible to expose a switch via Homebridge that would turn recording on/off via the Unifi API? Local storage is one of the key features of the Unifi Protect ecosystem so I suspect a lot of Unifi Protect users would be able to use such automation.

**Comment by hjdhjd** [comment]
First part: your welcome.

Second part: Huh? I have no idea what you're asking. Nothing about those sentences make sense to me. Can you rephrase and explain in simple, direct terms, how the feature option I already implemented doesn't do this.

**Comment by Ghazgkull** [comment]
Maybe what was added in v5.2.0 does add the feature requested here and I'm just misunderstanding. So let me validate my assumption... Looking at the documentation updates you pushed, I see a new option, `Enable.Video.HKSV.Recording.Switch`. Reading the description, it looks like this is an option which allows us to toggle recording on/off for Homekit Secure Video... Apple's encrypted cloud storage solution which I understand this plugin recently added support for (very cool).

If my understanding is correct, v5.2.0 doesn't address this use-case because this use-case isn't related to Homekit Secure Video. I don't want to just repeat the text of the issue description, but the idea here is to allow us to toggle recording on/off for a Unifi Protect camera via a switch exposed in Homekit. The end goal would be that when such a switch is toggled in Homekit, the Unifi camera's setting is updated to enable/disable recording.

**Comment by hjdhjd** [comment]
I understand your request better now. Your question was worded quite ambiguously. :smile:

Yes, I appreciate the use case...and yes, I'll add support for it in a future update. Thanks.

**Comment by Ghazgkull** [comment]
It was crystal clear and included links to relevant source code. But hey, we got there!

**Comment by hjdhjd** [comment]
It wasn't...it was crystal clear to you. Communication is **always** responsibility of the sender to be understood by the receiver.

Regardless...I'll update this issue once there's something to update. Thanks!

**Comment by hjdhjd** [comment]
Implemented in v5.4.0.

**Comment by Ghazgkull** [comment]
I tried this with the v5.4.0 release, but encountered a couple problems. The first was that the switch to toggle the recording on/off in Homekit seemed to not work. The switch shows as ON and I can try to toggle it OFF, but it just immediately flips back to the ON state.

The other problem I encountered is that just by enabling this switch, the LED recording lights on my cameras all turned on. So it looks like there might be a defect in the API where it's modifying other camera settings.

I will try the v5.5.0 release later today to see if either of those problems persist.

**Comment by hjdhjd** [comment]
There is no defect in the API. The defect is in HomeKit, and this is documented behavior in the HKSV documentation on this site. You can review that for more details.

You can't switch OFF one of the recording switches. You have to choose between one of the **three** states that are available, and it will adjust the other switches as appropriate. You only ever select ON one of the states. One of the three states will always be on. You get to select which one it is.

TL;DR: stop trying to turn things off. :smile: Just turn on the one you want and it will do the rest.
